### DISCUSSION TITLE
Circuit (codeforces ödevi)

### DISCUSSION THREAD

**Discussion by Ahmet2938** [sonsuzus/sonsuzus#484] [Soru Cevap]
A. Devre
test başına zaman sınırı1 saniye
test başına bellek sınırı256 megabayt

Alice az önce bir devre tasarlar ışıklar ve 2 günlük anahtarlar. Her bileşenin (bir ışık veya bir anahtar) iki durumu vardır: açık veya kapalı. Işıklar ve anahtarlar şu şekilde düzenlenmiştir:

Her ışık tam olarak iki anahtara bağlıdır .
Her anahtar tam olarak bir ışığa bağlı . Her anahtarın hangi ışığa bağlı olduğu bilinmiyor .
Bütün anahtarlar kapalı olduğunda bütün ışıklar da kapalı olur.
Bir anahtar açılıp kapatıldığında (açık konumdan kapalı konuma veya tam tersi), ona bağlı olan ışığın durumu da değişecektir.
Alice, yalnızca durumlarını gösteren devreyi getirir.2 gün
anahtarları kız kardeşi Iris'e verir ve ona bir bilmece sorar: En az ve en fazla kaç ışık açılabilir?

Küçük kız kardeşinin maskaralıklarını çok iyi bilen Iris, Alice'e doğru cevabı vermek için bir saniyeden fazla zaman harcamıyor. Siz de aynısını yapabilir misiniz?

Giriş
Her test birden fazla test durumundan oluşur. İlk satır tek bir tamsayı içerir
(1 ≤ t ≤ 500) — test vakalarının sayısı. Test vakalarının açıklaması aşağıdadır.

Her test durumunun ilk satırı tek bir tam sayı içerir
(1 ≤ n ≤ 50) — devredeki ışık sayısı.

Her test vakasının ikinci satırı şunları içerir:2 gün
tam sayılarA1,A2, … ,A2 gün
(0 ≤ABen≤ 1) — devredeki anahtarların durumları.ABen= 0
anlamına gelir Ben
-inci anahtar kapalıdır veABen= 1
anlamına gelir Ben
-inci anahtar açık.

Çıktı
Her test durumu için, sırasıyla açılabilecek minimum ve maksimum ışık sayısını belirten iki tam sayı çıktısı verin.
GİRİŞ :
5
1
0 0
1
0 1
1
1 1
3
0 0 1 0 1 0
3
0 1 1 1 0 0

ÇIKTI :
0 0
1 1
0 0
0 2
1 3

Not
İlk test durumunda devrede sadece bir ışık vardır ve hiçbir anahtar açık değildir, dolayısıyla ışık kesinlikle kapalıdır.

İkinci test durumunda ise devrede sadece bir adet ışık var, ancak bu ışığa bağlı bir anahtar açık olduğundan ışık yanıyor.

Üçüncü test durumunda ise devrede sadece bir ışık var ve her iki anahtar da açık olduğundan ışık iki kez açılıp kapatıldığı için kapalı durumdadır.

Dördüncü test durumunda, hiçbir ışığın yanmaması için anahtarlar şu şekilde düzenlenebilir:

Anahtar1
ve geçiş4
ışığa bağlıdır1
Her iki anahtar da kapalı olduğundan ışık1
Ayrıca kapalıdır.
Anahtar2
ve geçiş6
ışığa bağlıdır2
Her iki anahtar da kapalı olduğundan ışık2
Ayrıca kapalıdır.
Anahtar3
ve geçiş5
ışığa bağlıdır3
Her iki anahtar da açık, bu yüzden hafif3
başlangıçtaki kapalı durumundan iki kez açılıp kapatılır ve böylece kapalı kalır.
Ve sahip olmak2
Işıklar açıkken, anahtarlar şu şekilde düzenlenebilir:

Anahtar1
ve geçiş2
ışığa bağlıdır1
Her iki anahtar da kapalı olduğundan ışık1
Ayrıca kapalıdır.
Anahtar3
ve geçiş4
ışığa bağlıdır2
. Anahtardan beri3
açık ve anahtar4
kapalı, ışık2
başlangıçtaki kapalı durumundan bir kez açılıp kapatıldığında açıktır.
Anahtar5
ve geçiş6
ışığa bağlıdır3
. Anahtardan beri5
açık ve anahtar6
kapalı, ışık3
başlangıçtaki kapalı durumundan bir kez açılıp kapatıldığında açıktır.

**Comment by Ahmet2938** [comment]
```python
def cozum():
    girdi = input("Test sayısını girin: ")
    t = int(girdi)
    
    cikti = []
    
    for i in range(t):
        n = int(input(f"{i+1}. test için ışık sayısını girin: "))
        anahtarlar = list(map(int, input("Anahtarların durumunu girin (0 veya 1): ").split()))        
        toplam_acik = sum(anahtarlar)
        
        max_isik = min(n, toplam_acik // 2)
        min_isik = 1 if toplam_acik > 0 else 0
        
        cikti.append(f"{min_isik} {max_isik}")
    print("\n".join(cikti))

cozum()
```

**Reply by sonsuzus** [reply]
Soruyu daha anlaşılır bir şekilde yazarsan (ve anlarsan) çözümü de kolay olacaktır.
